src/main/java/org/pawaneuler/DataTypes/Rule/Frequency.java

```diff
@@ -0,0 +1,100 @@
+package org.pawaneuler.DataTypes.Rule;
+
+import java.util.ArrayList;
+
+import org.pawaneuler.DataTypes.Trie.Node;
+import org.pawaneuler.DataTypes.Trie.Trie;
+
+public class Frequency {
+    private Trie T;
+    private ArrayList<String> itemset;
+    private int allFreq;
+
+    public Frequency(Trie T) {
+        this.T = T;
+    }
+
+    public int generateAllFreq() {
+        ArrayList<Node> nodes = T.getNodes();
+        int sum = 0;
+        for (int i = 0; i < nodes.size(); i++) {
+            sum += T.getNodeAt(i).getFrequency();
+        }
+        return sum;
+    }
+
+    public int getAllFreq() {
+        return allFreq;
+    }
+
+    public int getItemsetFreq(ArrayList<String> itemset) {
+        this.itemset = itemset;
+        return getItemsetFreqRecursive(0, 0);
```

Comment:
A little context wouldn't hurt.

```suggestion
        // Start at first item and first node of the Trie
        return getItemsetFreqRecursive(0, 0);
```